hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -1200,13 +1200,24 @@ public FileStatus getFileStatus(Path f) throws IOException {
         INode<FileSystem> inode = iEntry.getValue();
         if (inode.isLink()) {
           INodeLink<FileSystem> link = (INodeLink<FileSystem>) inode;
-
-          result[i++] = new FileStatus(0, false, 0, 0,
-            creationTime, creationTime, PERMISSION_555,
-            ugi.getShortUserName(), ugi.getPrimaryGroupName(),
-            link.getTargetLink(),
-            new Path(inode.fullPath).makeQualified(
-                myUri, null));
+          try {
+            FileStatus status = link.getTargetFileSystem()
+                .getFileStatus(new Path("/"));
```

Comment:
Why are we getting status on "/"? In practical scenario it should work. However what if the target uri is a file?
Should we simply use link.getTargetFileSystem().getUri() ?
Could you please check scenario? If this works, I have no other changes.
Thanks for update.

Reply:
Changed to link.getTargetFileSystem().getUri(). Used ChRootedFileSystem to obtain the filestatus. `((ChRootedFileSystem)link.getTargetFileSystem()).getMyFs().getFileStatus()`.